records/2025-02-19.authmethods.md

```diff
@@ -0,0 +1,174 @@
+3. We would then disable 2FA verification by SMS for all Notify users.  
+
+Benefits of removing SMS-based 2FA:  
+• Eliminates vulnerabilities to SIM swapping attacks  
+• SMS is known to be the least secure form of 2FA
+• Improves overall security posture
+
+
+## 2. Mobile based Authenticator Apps (TOTP - Time based one time password)
+Authenticator apps are widely used by multiple clients and teams for signup flows. Most users will have encountered them while using everyday banking or mail applications.
+Authenticators provide one-time passwords generated by applications like Google Authenticator, Microsoft Authenticator, or Authy.
+Introducing an authenticator application will not solve the issue of users who do not have a cell phone provided by their department.
+This is just an additional 2FA method for clients with a department-provided cellphone.
+
+### Benefits:  
+• Works offline once configured  
+• Not vulnerable to SIM swapping  
+• No SMS costs  
+• Higher security than SMS  
+
+### Cons:  
+• Requires smartphone and app installation  
+• Device loss can lead to account recovery challenges  
+• Slightly higher friction during initial setup  
+
+### Implementation:    
+The FIDO2 library that Notify currently uses for implementing YubiKey can be extended to include Authenticator apps.
+The implementation would follow the details for YubiKey and store login events in the sessions table.
+
+## 3. [Magic Link ADR](https://github.com/cds-snc/notification-adr/pull/64/files)
+We can revive the work done in the above ADR for magic links. There is a stock implementation in Notify, we could use for Magic Links.
+Please see the above ADR for introducing magic links into our 2FA flow
+
+## 4. Default to using email for 2FA
+If we remove SMS as a 2FA method, we should default the 2FA to email authentication.  
+• Requires minimal changes to authentication framework as this is already implemented  
+• Verification logic remains largely unchanged from SMS implementation  
+
+# Decision
+
+We are not going to offer SMS 2FA auth going forward. There are two reasons for this:  
+1. SMS 2FA is not secure (as shown in recommendation 1)
+2. Clients without a phone should still be able to sign up for notify
+
+Even though this ADR isn't meant to capture login flow - login flow is crucial to determining how a user sets up their 2FA for the first time.
+On the profile page, a user should be able to choose any of the recommended auth options. The design of the page should be such that a user chooses the following 2FA options in order:  
+1. YubiKey
+2. Authenticator app
+3. Magic link
+4. Email code auth
+
+Recognizing that the second option is mobile-based(see section below for Desktop based Authenticator apps), we are still going to offer authenticator apps for clients who have smart devices.
+
+**If a client doesn't have a yubi key or a phone, we will default to email_auth for them.**
+
+## Other options that we can revisit in the future
```

Comment:
I would just move these into the options section (and mark these as rejected), or get rid of the section from this ADR to reduce noise.